support/src/main/java/org/jenkinsci/plugins/workflow/support/pickles/FilePathPickle.java

```diff
@@ -86,6 +86,18 @@ protected FilePath tryResolve() {
         // TODO better to use a synchronized accessor
         @Restricted(NoExternalUse.class)
         public static final Map<VirtualChannel,String> channelNames = new WeakHashMap<VirtualChannel,String>();
+
+        static {
+            // Add channels that may already be known to Jenkins. This can be the case
```

Comment:
Alternate idea (might be more robust): encapsulate the lookups of `channelNames` (in `WorkspaceActionImpl`, `WorkspaceListLeasePickle`, and `FilePathPickle` itself) into a method which before returning `null` (or throwing `IllegalStateException`, as all currently impls do) would fall back to doing this inspection of the currently online `Computer`s.